pyproject.toml

```diff
@@ -18,7 +18,7 @@ dynamic = ["version"]
 dependencies = [
     "pandas>=1.5.0",
     "scikit-learn>=1.2.0",
-    "narwhals >= 1.9.2",
+    "narwhals >= 1.9.4",
```

Comment:
N: In future, could you include the rationale for version bumps in the git commit message? 'env update pyarrow' Doesn't help me understand why the point version had to be increase.

Reply:
Do we need this version and up only? It's [very recent](https://pypi.org/project/narwhals/#history).

I don't think that we should change this in our PR here, but a separate piece of work is to define a strategy around API versions. narwhals has a [Perfect backwards compatibility policy,](https://narwhals-dev.github.io/narwhals/backcompat/) which feels like something to make use of.

Reply:
fair point - re narwhals, we do actually need this one upwards as it's where they actioned this bug for us https://github.com/narwhals-dev/narwhals/issues/1200